miden-lib/src/notes/mod.rs

```diff
@@ -33,33 +35,25 @@ pub fn create_note(
     serial_num: Word,
 ) -> Result<Note, NoteError> {
     let note_assembler = assembler();
-    let (note_script, inputs): (&str, Vec<Felt>) = match script {
-        Script::P2ID { target } => ("p2id", vec![target.into(), ZERO, ZERO, ZERO]),
+
+    // Include the binary version of the scripts into the source file at compile time
+    let p2id_bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/P2ID.masb"));
+    let p2idr_bytes = include_bytes!(concat!(env!("OUT_DIR"), "/assets/P2IDR.masb"));
+
+    let (note_script_ast, inputs): (ProgramAst, Vec<Felt>) = match script {
+        Script::P2ID { target } => (
+            ProgramAst::from_bytes(p2id_bytes).unwrap(),
+            vec![target.into(), ZERO, ZERO, ZERO],
+        ),
         Script::P2IDR {
             target,
             recall_height,
-        } => ("p2idr", vec![target.into(), recall_height.into(), ZERO, ZERO]),
+        } => (
+            ProgramAst::from_bytes(p2idr_bytes).unwrap(),
+            vec![target.into(), recall_height.into(), ZERO, ZERO],
+        ),
     };
```

Comment:
Instead of `unwrap()` - should we return errors in case parsing of scripts fails?

Reply:
I now return `ProgramAst::from_bytes(p2id_bytes).expect("Cannot deserialize the note script")`

I would be better to create a new `NoteError` that passes on the `DeserializationError`. But for that we need to add `Clone` to 

https://github.com/facebook/winterfell/blob/main/utils/core/src/errors.rs

Let me know, if you want to do this.

Reply:
Yes, better return a `NoteError`. It could contain just a string (and we should be able to reduce `DeserializationError` to string. But `DeserializationError` is now `Clone` in Winterfell - so, if you do `cargo update`, you should be able to include it in the `NoteError`.